jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/RuntimeGroup.kt

```diff
@@ -139,38 +97,24 @@ fun AnActionEvent.runtime(): Runtime? {
 }
 
 /**
- * A bean that represents an extension point based on a [RuntimeGroup]
+ * To be implemented on a companion object of the extension point object to expose factory methods.
+ * See [software.aws.toolkits.jetbrains.services.lambda.LambdaBuilder]
  */
-class RuntimeGroupExtensionPoint<T> : AbstractExtensionPointBean(), KeyedLazyInstance<T> {
+abstract class RuntimeGroupExtensionPointObject<T>(private val extensionPointName: ExtensionPointName<IdBasedExtensionPoint<T>>) {
+    private val collector = KeyedExtensionCollector<T, String>(extensionPointName.name)
 
-    @Attribute("implementation")
-    lateinit var implementation: String
+    fun getInstance(runtimeGroup: RuntimeGroup): T? = collector.findSingle(runtimeGroup.id)
+    fun getInstanceOrThrow(runtimeGroup: RuntimeGroup): T = getInstance(runtimeGroup)
```

Comment:
Same comment about making it `getInstance`/`getInstanceOrNull`

Reply:
Agree, we should prioritize `getInstanceOrThrow` since that is what we use more